### PR TITLE
Checking service parameters UnitTests

### PR DESCRIPTION
**Pull request by petko-boyadzhiev**

### CODE REVIEW

File: com.sap.cloud.lm.sl.cf.process/src/test/java/com/sap/cloud/lm/sl/cf/process/steps/DetermineServiceCreateUpdateServiceActionsStepTest.java [hunk 1]
```diff
@@ -163,6 +165,8 @@ private void prepareClient() {
                 .thenReturn(existingService);
             Mockito.when(client.getServiceInstance(stepInput.existingService.getName(), false))
                 .thenReturn(existingServiceInstance);
+            Mockito.when(client.getServiceParameters(existingService.getMetadata().getGuid()))
```
Comment (boyan-velinov) [inline com.sap.cloud.lm.sl.cf.process/src/test/java/com/sap/cloud/lm/sl/cf/process/steps/DetermineServiceCreateUpdateServiceActionsStepTest.java]: Why there is not positive test where service parameters are present?

File: com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/DetermineServiceCreateUpdateServiceActionsStep.java [hunk 2]
```diff
@@ -255,7 +255,7 @@ private boolean shouldUpdateCredentials(CloudServiceExtended service, CloudServi
             if (HttpStatus.NOT_IMPLEMENTED == e.getStatusCode() || HttpStatus.BAD_REQUEST == e.getStatusCode()) {
                 getStepLogger().warnWithoutProgressMessage(e, Messages.CANNOT_RETRIEVE_SERVICE_PARAMETERS, service.getName());
                 // TODO: Optimization (Hack) that should be deprecated at some point. So here is a todo for that.
-                return MapUtils.isEmpty(service.getCredentials());
+                return !MapUtils.isEmpty(service.getCredentials());
```
Comment (theghost5800) [inline com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/DetermineServiceCreateUpdateServiceActionsStep.java]: Why we change this behaviour?

File: com.sap.cloud.lm.sl.cf.process/src/test/java/com/sap/cloud/lm/sl/cf/process/steps/DetermineServiceCreateUpdateServiceActionsStepTest.java (hunk 1 above)
Comment (petko-boyadzhiev) [reply com.sap.cloud.lm.sl.cf.process/src/test/java/com/sap/cloud/lm/sl/cf/process/steps/DetermineServiceCreateUpdateServiceActionsStepTest.java]: Done

File: com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/DetermineServiceCreateUpdateServiceActionsStep.java (hunk 2 above)
Comment (petko-boyadzhiev) [reply com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/DetermineServiceCreateUpdateServiceActionsStep.java]: Previous was wrong.

**Pull request merged**